src/wallet/rpcwallet.cpp

```diff
@@ -3289,6 +3291,47 @@ void FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& fee_out,
         coinControl.fAllowWatchOnly = ParseIncludeWatchonly(NullUniValue, wallet);
     }
 
+    if (!solving_data.isNull()) {
+        if (solving_data.exists("pubkeys")) {
+            const UniValue pubkey_strs = solving_data["pubkeys"].get_array();
+            for (unsigned int i = 0; i < pubkey_strs.size(); ++i) {
```

Comment:
Nit: is there any reason not to use a C++-style loop here? It's used further on on line 3359 and so on.

Reply:
Done